src/field_theory/ratfunc.lean

```diff
@@ -220,19 +220,19 @@ end
 
 lemma lift_on_condition_of_lift_on'_condition {P : Sort v} {f : ∀ (p q : K[X]), P}
   (H : ∀ {p q a} (hq : q ≠ 0) (ha : a ≠ 0), f (a * p) (a * q) = f p q)
-  ⦃p q p' q' : K[X]⦄ (hq : q ≠ 0) (hq' : q' ≠ 0) (h : p * q' = p' * q) :
+  ⦃p q p' q' : K[X]⦄ (hq : q ≠ 0) (hq' : q' ≠ 0) (h : q' * p = q * p') :
   f p q = f p' q' :=
 begin
   have H0 : f 0 q = f 0 q',
   { calc f 0 q = f (q' * 0) (q' * q) : (H hq hq').symm
            ... = f (q * 0) (q * q') : by rw [mul_zero, mul_zero, mul_comm]
            ... = f 0 q' : H hq' hq },
   by_cases hp : p = 0,
-  { simp only [hp, hq, zero_mul, or_false, zero_eq_mul] at ⊢ h, rw [h, H0] },
+  { simp only [hp, hq, mul_zero, false_or, zero_eq_mul] at ⊢ h, rw [h, H0] },
   by_cases hp' : p' = 0,
-  { simpa only [hp, hp', hq', zero_mul, or_self, mul_eq_zero] using h },
+  { simpa only [hp, hp', hq', mul_zero, or_self, mul_eq_zero] using h },
   calc f p q = f (p' * p) (p' * q) : (H hq hp').symm
-         ... = f (p * p') (p * q') : by rw [mul_comm p p', h]
+         ... = f (p * p') (p * q') : by rw [mul_comm p p', mul_comm _ q, ←h, mul_comm p]
          ... = f p' q' : H hq' hp
 end
 
```

Comment:
Turns out only the last three lines are necessary:
```suggestion
calc f p q = f (q' * p) (q' * q) : (H hq hq').symm
       ... = f (q * p') (q * q') : by rw [h, mul_comm q']
      ... = f p' q' : H hq' hq
```
BTW, I don't know what the `-- f` immediately below this means.